Use cmd version of eslint and enable caching

mramato opened the pull request:
While reviewing #5469 I came across what I feel is a better way to  manage our linting which has less boilerplate and I feel is easier to manage.

@omh1280 can you review this since it supercedes #5469.  Thanks.

Full changelog.

1. Caching makes eslint only take ~3 seconds plus any files that have changed since the last time you ran it. Since it's unlikely devs are touching every tile between runs, this makes eslint much incredibly faster in the average case.  Also added the genereated `.eslintcache` to git ignore.

2. Switched to the pure cli version of eslint and remove `eslint-watch`, which I'm pretty sure no one uses anyway since it was so impractical at ~30 seconds a run. This simplified our usage and means we lint all js and html files by default except for the globs specifically listed in `.eslintignore`  This also shaves 2-4 seconds off startup time because we're not loading gulpfile.js anymore.

3. Fixed an issue in `index.release.html`, which was previously not linted.

omh1280 commented on package.json:
```diff
@@ -83,8 +83,7 @@
     "instrumentForCoverage": "gulp instrumentForCoverage",
-    "eslint": "gulp eslint",
-    "eslint-watch": "gulp eslint-watch",
+    "eslint": "eslint ./**/*.js ./**/*.html --cache",
```
Maybe add `--quiet` to suppress warnings? I think that was the `gulp-eslint` default

mramato replied:
Unless I'm mistaken, `--quiet` would actually not be something we want.  We want to see any and all problems reported by `eslint`.  If quiet was the default before, then that's an oversight on my part.

omh1280 replied:
Well, quiet just suppresses warnings; errors go through.

omh1280 replied:
Like, the warning `File ignored because of a matching ignore pattern. Use "--no-ignore" to override` shouldn't fail travis

mramato replied:
OK, I misunderstood what "warnings" were.  I thought there was a class of eslint rules but it looks like it's just eslint being all "are you sure".  I don't see the warning you're describing locally, but I do on travis.  I'll add the --quite flag.

mramato commented:
I squashed my changes and added quite.  Once this is green we should be able to merge.

@lilleyse or @tfili can one of you do a quick review and actual merge since @omh1280 doesn't have commit access?

Thanks.

tfili commented:
Looks fine to me. I'll merge when its green.
